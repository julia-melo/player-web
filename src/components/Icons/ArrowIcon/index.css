.arrow-up-icon svg,
.arrow-down-icon svg,
.arrow-left-icon svg,
.arrow-right-icon svg {
  display: block;
}

.arrow-up-icon {
  transform: rotate(270deg);
}

.arrow-down-icon {
  transform: rotate(90deg);
}

.arrow-left-icon {
  transform: rotate(180deg);
}

.arrow-up-icon.clickable,
.arrow-down-icon.clickable,
.arrow-left-icon.clickable,
.arrow-right-icon.clickable {
  cursor: pointer
}

.arrow-up-icon.disabled,
.arrow-down-icon.disabled,
.arrow-left-icon.disabled,
.arrow-right-icon.disabled {
  opacity: 0.4;
  cursor: auto
}
