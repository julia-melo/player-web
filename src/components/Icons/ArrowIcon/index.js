import React from 'react'
import PropTypes from 'prop-types'
import { ReactSVG } from 'react-svg'

import arrowIcon from '../../../assets/arrow.svg'

import './index.css'

const ArrowIcon = ({
  className,
  width = 80,
  height = 80,
  disabled = false,
  onClick
}) => (
  <ReactSVG
    className={`${className} ${disabled ? 'disabled' : ''} ${onClick ? 'clickable' : ''}`}
    src={arrowIcon}
    beforeInjection={svg => {
      svg.setAttribute('style', `width: ${width}px; height: ${height}px;`)
    }}
    onClick={onClick}
  />
)

ArrowIcon.propTypes = {
  className: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
}

const ArrowRightIcon = props => (
  <ArrowIcon className='arrow-right-icon' {...props} />
)

const ArrowLeftIcon = props => (
  <ArrowIcon className='arrow-left-icon' {...props} />
)

const ArrowUpIcon = props => (
  <ArrowIcon className='arrow-up-icon' {...props} />
)

const ArrowDownIcon = props => (
  <ArrowIcon className='arrow-down-icon' {...props} />
)

export { ArrowUpIcon, ArrowDownIcon, ArrowLeftIcon, ArrowRightIcon }
