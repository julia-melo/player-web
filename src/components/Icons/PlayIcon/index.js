import React from 'react'
import PropTypes from 'prop-types'
import { ReactSVG } from 'react-svg'

import playIcon from '../../../assets/play.svg'

const PlayIcon = ({ width = 80, height = 80 }) => (
  <ReactSVG
    className='play-icon'
    src={playIcon}
    beforeInjection={svg => {
      svg.setAttribute('style', `width: ${width}px; height: ${height}px`)
    }}
  />
)

PlayIcon.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
}

export default PlayIcon
