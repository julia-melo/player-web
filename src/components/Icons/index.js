export * from './ArrowIcon'
export { default as PlayIcon } from './PlayIcon'
