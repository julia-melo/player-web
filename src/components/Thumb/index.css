.thumb {
  width: 100%;
  height: 100%;

  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  cursor: pointer;
}

.thumb .content {
  display: flex;
  align-items: center;
  justify-content: center;
  
  transition: opacity .2s;
  opacity: 0.6;
}

.thumb:hover .content {
  opacity: 0.8;
}

.thumb .content .play-icon {
  position: absolute;
  z-index: 1;
}

.thumb .content .image {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.thumb .title {
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}
