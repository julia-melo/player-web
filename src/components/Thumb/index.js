import React from 'react'
import PropTypes from 'prop-types'

import { PlayIcon } from '../Icons'

import './index.css'

const Thumb = ({ image, size, caption }) => {
  const contentStyle = size ? { width: `${size}px`, height: `${size/1.6}px` } : {}
  const titleStyle = size ? { width: `${size}px` } : {}
  const iconSize = size && size / 3

  return (
    <div className='thumb'>
      <div
        className='content'
        style={contentStyle}
      >
        <img className='image' src={image} alt='thumb' />
        <PlayIcon width={iconSize} height={iconSize} />
      </div>
      {caption && <p className='title' style={titleStyle}>{caption}</p>}
    </div>
  )
}

Thumb.propTypes = {
  image: PropTypes.string.isRequired,
  size: PropTypes.number,
  caption: PropTypes.string,
}

export default Thumb
