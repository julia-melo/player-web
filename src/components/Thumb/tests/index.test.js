import React from 'react'
import renderer from 'react-test-renderer'

import { PlayIcon } from '../../Icons'
import Thumb from '../index'

const mockImage = 'https://designshack.net/wp-content/uploads/placeholder-image.png'

describe('Thumb component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Thumb image={mockImage} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('receives size prop', () => {
    const instance = renderer
      .create(<Thumb image={mockImage} size={48} />)
      .root

    expect(instance.findByType(PlayIcon).props.width).toBe(16)
    expect(instance.findByProps({className: 'content'}).props.style.width).toBe('48px')
    expect(instance.findByProps({className: 'content'}).props.style.height).toBe('30px')
  })

  it('receives caption prop', () => {
    const exampleCaption = 'Example caption'
    const instance = renderer
      .create(<Thumb image={mockImage} caption={exampleCaption} />)
      .root

    expect(instance.findByProps({className: 'title'}).children[0]).toBe(exampleCaption)
  })
})
