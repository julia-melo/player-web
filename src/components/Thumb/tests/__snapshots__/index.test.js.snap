// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumb component renders correctly 1`] = `
<div
  className="thumb"
>
  <div
    className="content"
    style={Object {}}
  >
    <img
      alt="thumb"
      className="image"
      src="https://designshack.net/wp-content/uploads/placeholder-image.png"
    />
    <div
      className="play-icon"
    />
  </div>
</div>
`;
